src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java

```diff
@@ -452,104 +449,99 @@ public SolrQueryResponse search(DataverseRequest dataverseRequest, Dataverse dat
+                     * "dataverses". Also, is an API token required for published
+                     * dataverses? Michael: url changed.
+                     */
 //                solrSearchResult.setApiUrl(baseUrl + "/api/dataverses/" + entityid);
-            } else if (type.equals("datasets")) {
-                solrSearchResult.setHtmlUrl(baseUrl + "/dataset.xhtml?globalId=" + identifier);
-                solrSearchResult.setApiUrl(baseUrl + "/api/datasets/" + entityid);
-                //Image url now set via thumbnail api
-                //solrSearchResult.setImageUrl(baseUrl + "/api/access/dsCardImage/" + datasetVersionId);
-                // No, we don't want to set the base64 thumbnails here. 
-                // We want to do it inside SearchIncludeFragment, AND ONLY once the rest of the 
-                // page has already loaded.
-                //DatasetVersion datasetVersion = datasetVersionService.find(datasetVersionId);
-                //if (datasetVersion != null){                    
-                //    solrSearchResult.setDatasetThumbnail(datasetVersion.getDataset().getDatasetThumbnail(datasetVersion));
-                //}
-                /**
-                 * @todo Could use getFieldValues (plural) here.
-                 */
-                ArrayList<String> datasetDescriptions = (ArrayList<String>) solrDocument.getFieldValue(SearchFields.DATASET_DESCRIPTION);
-                if (datasetDescriptions != null) {
-                    String firstDatasetDescription = datasetDescriptions.get(0);
-                    if (firstDatasetDescription != null) {
-                        solrSearchResult.setDescriptionNoSnippet(firstDatasetDescription);
+                    break;
+                case "datasets":
+                    solrSearchResult.setHtmlUrl(baseUrl + "/dataset.xhtml?globalId=" + identifier);
+                    solrSearchResult.setApiUrl(baseUrl + "/api/datasets/" + entityid);
+                    //Image url now set via thumbnail api
+                    //solrSearchResult.setImageUrl(baseUrl + "/api/access/dsCardImage/" + datasetVersionId);
+                    // No, we don't want to set the base64 thumbnails here.
+                    // We want to do it inside SearchIncludeFragment, AND ONLY once the rest of the
+                    // page has already loaded.
+                    //DatasetVersion datasetVersion = datasetVersionService.find(datasetVersionId);
+                    //if (datasetVersion != null){
+                    //    solrSearchResult.setDatasetThumbnail(datasetVersion.getDataset().getDatasetThumbnail(datasetVersion));
+                    //}
+                    /**
+                     * @todo Could use getFieldValues (plural) here.
+                     */
+                    ArrayList<String> datasetDescriptions = (ArrayList<String>) solrDocument.getFieldValue(SearchFields.DATASET_DESCRIPTION);
```

Comment:
This one isn't possible, trying to gives `Collection<Object> cannot be converted to List<String>` (note the typo)